aws/resource_aws_workspaces_workspace_test.go

```diff
@@ -0,0 +1,578 @@
+	conn := testAccProvider.Meta().(*AWSClient).workspacesconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_workspace" {
+			continue
+		}
+
+		resp, err := conn.DescribeWorkspaces(&workspaces.DescribeWorkspacesInput{
+			WorkspaceIds: []*string{aws.String(rs.Primary.ID)},
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(resp.Workspaces) == 0 {
+			return fmt.Errorf("workspace %q was not terminated", rs.Primary.ID)
+		}
+		ws := resp.Workspaces[0]
+
+		if *ws.State != workspaces.WorkspaceStateTerminating && *ws.State != workspaces.WorkspaceStateTerminated {
+			return fmt.Errorf("workspace %q was not terminated", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsWorkspacesWorkspaceExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).workspacesconn
+
+		_, err := conn.DescribeWorkspaces(&workspaces.DescribeWorkspacesInput{
+			WorkspaceIds: []*string{aws.String(rs.Primary.ID)},
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccAwsWorkspacesWorkspaceConfig_Prerequisites() string {
+	return fmt.Sprintf(`
+data "aws_region" "current" {}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+locals {
+  region_workspaces_az_ids = {
+    "us-east-1" = formatlist("use1-az%%d", [2, 4, 6])
+  }
+
+  workspaces_az_ids = lookup(local.region_workspaces_az_ids, data.aws_region.current.name, data.aws_availability_zones.available.zone_ids)
+}
+
+data "aws_vpc" "default" {
+  default = true
+}
+
+data "aws_subnet" "default" {
+  count = length(local.workspaces_az_ids)
+  availability_zone_id = "${local.workspaces_az_ids[count.index]}"
+  default_for_az = true
+}
+
+resource "aws_directory_service_directory" "test" {
+  size = "Small"
+  name = "tf-acctest.neverland.com"
+  password = "#S1ncerely"
+
+  vpc_settings {
+    vpc_id = "${data.aws_vpc.default.id}"
+	subnet_ids = ["${data.aws_subnet.default.*.id[0]}", "${data.aws_subnet.default.*.id[1]}"]
+  }
+}
+
+data "aws_iam_policy_document" "workspaces" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["workspaces.amazonaws.com"]
+    }
+  }
+}
+
+# resource "aws_iam_role" "workspaces-default" {
+#   name               = "workspaces_DefaultRole"
+#   assume_role_policy = data.aws_iam_policy_document.workspaces.json
+# }
+# 
+# resource "aws_iam_role_policy_attachment" "workspaces-default-service-access" {
+#   role       = aws_iam_role.workspaces-default.name
+#   policy_arn = "arn:aws:iam::aws:policy/AmazonWorkSpacesServiceAccess"
+# }
+# 
+# resource "aws_iam_role_policy_attachment" "workspaces-default-self-service-access" {
+#   role       = aws_iam_role.workspaces-default.name
+#   policy_arn = "arn:aws:iam::aws:policy/AmazonWorkSpacesSelfServiceAccess"
+# }
```

Comment:
If these aren't needed, we can just remove them

Reply:
Sorry, that's leaked from git stash. I have to comment out this section every time I run acctests, because the target account already have this role in place. I've compared the current workspaces default role policy attachments and found some changed since I started this PR:

- `AmazonWorkSpacesServiceAccess` –> `SkyLightServiceAccess`
- `AmazonWorkSpacesSelfServiceAccess` –> `SkyLightSelfServiceAccess`

Fixed.

Reply:
If you are curious as me what does `SkyLight` mean – it's a AWS workspace agent written in Go.